### DISCUSSION TITLE
Plugin-side command to create MinecraftServer

### DISCUSSION THREAD

**Discussion by DiogoMurano** [jeremylvln/Shulker#647] [Ideas]
Should it be possible to create MinecraftServer from a command or API integration using shulker?
Is this a feature that makes sense considering Shulker's purpose?

**Comment by jeremylvln** [comment]
You can already embed the Shulker Java SDK in your projects, it currently only have a method summon a server, as you suggested. It directly interacts with Shulker controller. I agree it would be cool to also add a command as well to simplify this use-case!

**Reply by DomiIRL** [reply]
Hey @jeremylvln, a powerful SDK would be really wonderful. 
I would like to create some plugins that also need information that currently only the proxy api provides on the minecraft servers for example and also need to manage existing game servers in regards to starting and stopping them aswell as adding volumes when starting etc. Is there currently a way to do these things?

**Reply by jeremylvln** [reply]
Hi @DomiIRL, I only created SDK methods I'm using myself, but you are totally right: the objective is to have a complete SDK which can serve all kind of functions.

For now, I really want to focus on the declarative way of provisioning the infrastructure, through the CRDs. I find it a bit weird to have methods to stop game servers (whereas there is already a method to summon a new server).

Do not hesitate to list the features you are interested in, with some use-cases examples. So we can discuss together whether it's meaningful to add support for or not. It's really important to me to not fall into the pitfall of adding support for everything and finally have a software that is unmaintainable!

**Reply by jeremylvln** [reply]
I'm just closing this discussion because the original request is now resolved, but please create a new one so we can discuss about adding features to the proxy/server's APIs and/or Shulker's SDK!
